pkg/api/handlers/compat/containers_top.go

```diff
@@ -58,7 +59,24 @@ loop: // break out of for/select infinite` loop
 		case <-r.Context().Done():
 			break loop
 		default:
-			output, err := c.Top(strings.Split(query.PsArgs, ","))
+			// Note that the descriptors to ps(1) must be shlexed (see #12452).
+			psDescriptors := []string{}
+			shSplit, err := shlex.Split(query.PsArgs)
+			if err != nil {
+				utils.InternalServerError(w, err)
+				break loop
+			}
+			for _, s := range shSplit {
+				if s != "" {
+					if strings.Contains(s, " ") {
+						s = fmt.Sprintf("%q", s)
+					}
+					psDescriptors = append(psDescriptors, s)
+				}
+			}
```

Comment:
This feels and looks incorrect, the proper fix should be to remove the shlex.Split call in c.Top() in libpod

Reply:
Currently there are lots of use cases that are broken in the local and remote that are valid for ps command.  The code changes are all about getting the commands correct that we pass to the ps command.